common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -409,6 +466,131 @@ bool mon_wlan_hal_dwpal::sta_link_measurements_11k_request(const std::string &st
+
+    // prepare new scan params with changed dwell time
+    memcpy_s(&new_fg, sizeof(new_fg), &fg, sizeof(fg));
+    memcpy_s(&new_bg, sizeof(new_bg), &bg, sizeof(bg));
+    new_fg.passive = dwell_time_msec;
+    new_fg.active  = dwell_time_msec;
+    new_bg.passive = dwell_time_msec;
+    new_bg.active  = dwell_time_msec;
+
+    DWPAL_SET_SCAN_PARAMS_HELPER_FG(ifname, fg, new_fg);
+    DWPAL_GET_SCAN_PARAMS_HELPER_FG(ifname, out_data, out_len, NL_MAX_REPLY_BUFFSIZE, new_fg);
+
+    DWPAL_SET_SCAN_PARAMS_HELPER_BG(ifname, bg, new_bg);
+    DWPAL_GET_SCAN_PARAMS_HELPER_BG(ifname, out_data, out_len, NL_MAX_REPLY_BUFFSIZE, new_bg);
+
+    if ((new_fg.active != dwell_time_msec) && (new_fg.passive != dwell_time_msec) &&
+        (new_bg.active != dwell_time_msec) && (new_bg.passive != dwell_time_msec)) {
+        LOG(ERROR) << __func__ << " failed to set dwell time =" << dwell_time_msec << std::endl
+                   << "new fg: " << new_fg.passive << " " << new_fg.active << " "
+                   << new_fg.num_probe_reqs << " " << new_fg.probe_reqs_interval << " "
+                   << new_fg.passive_scan_valid_time << " " << new_fg.active_scan_valid_time
+                   << std::endl
+                   << "new bg: " << new_bg.passive << " " << new_bg.active << " "
+                   << new_bg.num_probe_reqs << " " << new_bg.probe_reqs_interval << " "
+                   << new_bg.num_chans_in_chunk << " " << new_bg.break_time << " "
+                   << new_bg.break_time_busy;
+        return false;
+    }
+
+    LOG(DEBUG) << "dwell_time_msec updated from passive = " << fg.passive
+               << " active = " << fg.active << " to = " << dwell_time_msec;
+
+    int freq_index = 0;
+    //configure center frequency for each scanned channel
+    for (auto channel : channel_pool) {
+        //channel validation
+        LOG(DEBUG) << " validating pool channel=" << channel;
+        if (son::wireless_utils::which_freq(m_radio_info.channel) ==
+            beerocks::eFreqType::FREQ_24G) {
+            // 2.4G interface
+            if (son::wireless_utils::which_freq(channel) == beerocks::eFreqType::FREQ_5G) {
+                LOG(ERROR) << " cannot scan 5G channel=" << int(channel)
+                           << " on 2G interface=" << m_radio_info.iface_name;
+                return false;
+            }
+        } else {
+            // 5G interface
+            if (son::wireless_utils::which_freq(channel) == beerocks::eFreqType::FREQ_24G) {
+                LOG(ERROR) << " cannot scan 2G channel=" << int(channel)
+                           << " on 5G interface=" << m_radio_info.iface_name;
+                return false;
+            }
+        }
+
+        dcs_scan_params.freq[freq_index] = beerocks::utils::wifi_channel_to_freq(int(channel));
+        LOG(DEBUG) << " dcs scan pool add center frequency=" << dcs_scan_params.freq[freq_index];
+        freq_index++;
+    }
+
+    dcs_scan_params.ap_force = 1;
```

Comment:
Please add a comment explaining what is this setting is done here and why is it not the default.

Reply:
I can add "must as single wifi won't allow scan on ap without this flag" hope its clear. 
also can be changed to bool , will be much clearer.

Reply:
added it as a comment to explain why we set ap_force = 1